benchmarks/sir/sir.stan

```diff
@@ -66,6 +56,5 @@ model {
   for (n in 2:N_t)
     stoi_hat[n] ~ poisson(y[n - 1, 1] - y[n, 1]);
 
-  B_hat ~ lognormal(log(col(to_matrix(y), 4)), 0.15);
-  
+  B_hat ~ lognormal(log(y[, 4]), 0.15);  
```

Comment:
Can we also vectorise the for-loop for stoi_hat please? And get the naming right.. so "hat" is the mean always while the observed data is not "hat"?

The for loop probably needs storing for a temporary for the differenced y.

Thanks.

Reply:
This would be the vectorized code:

```
stoi[1] ~ poisson(y0[1] - y[1, 1]);
stoi[2:N_t] ~ poisson(y[1:(N_t - 1), 1] - y[2:N_t, 1]);
```

It doesn't work because `y[1:(N_t - 1), 1]` is an array. We could store that in a temporary, but I like the for loop better so I'm staying with a for loop.

Reply:
Aww can we keep the hats? If I take out the hats then the data is different from the new and the old model. That's just inconvenient.

Reply:
Ah it's easier to just do this than argue about it. I'll change the data. It adds an extra link to the new interface case study but I suppose we'll survive :P.

Reply:
Thanks for getting the hats replaced... the for loop is still the far worse way of writing this. What's so bad about

```stan
  {
   vector[N_t] y_diff;
    y_diff[1] = y0[1] - y[1, 1];
    for (n in 2:N_t)
        y_diff[n] = y[n - 1, 1] - y[n, 1];

   stoi ~ poisson(y_diff);
  }
```

This is a lot faster for sure.

... but... OK, I just saw that N_t is 20... so whatever you prefer (I just think that these codes should reflect good Stan coding practice).

Reply:
Yeah you're right this would be better. Fixed.